AtSheet/explainer.md

```diff
@@ -0,0 +1,225 @@
+
+Named imports may be renamed as part of this import process:
+
+```js
+import baz, { bar as renamed } from 'sheet.css' with { type: 'css' }
+```
+
+The default import may be omitted, importing only the named `@sheet`:
+
+```js
+import { bar } from 'sheet.css' with { type: 'css' }
+```
+
+Any of these `import` examples can be then used to set the `adoptedStyleSheets` attribute on a Shadow DOM node:
+
+```js
+import { bar } from 'sheet.css' with { type: 'css' }
+document.adoptedStyleSheets = [bar];
+shadowRoot.adoptedStyleSheets = [bar];
+```
+
+#### Performance
+
+This will be a performance-neutral feature, and use of it may allow for developers to reduce the number of network requests. We should ensure that multiple imports of different sheets from the same file produce a single network request. 
+
+```js
+// The following two imports should only make a single network request.
+import { foo } from 'sheet.css' with { type: 'css' };
+import { bar } from 'sheet.css' with { type: 'css' }
+```
+
+```html
+<style>
+/* The following two imports should only make a single network request. */
+@import "sheet.css#foo";
+@import "sheet.css#bar";
+</style>
+```
+
+```html
+<!-- The following two link tags should only make a single network request. -->
+<link rel="stylesheet" href="sheet.css#foo" />
+<link rel="stylesheet" href="sheet.css#bar" />
+```
+
+#### Interaction with CSSOM
+
+
+Named `@sheet` references augment the [existing](https://drafts.csswg.org/cssom/#stylesheet) `StyleSheet` interface with an optional `name` attribute reflecting the `@sheet` identifier:
+
+```
+[Exposed=Window]
+interface StyleSheet {
+  readonly attribute DOMString? name;
+};
+```
+*Open issue: Should this overload the existing `title` attribute instead?*
+
+This also expands the [existing](https://drafts.csswg.org/cssom/#cssstylesheet) CSSOM `CSSStyleSheet` definition with a `StyleSheetList` of nested `CSSStyleSheet` objects to access nested `@sheet` references:
+
+```
+[Exposed=Window]
+interface CSSStyleSheet : StyleSheet {
+  [SameObject] readonly attribute StyleSheetList nestedStyleSheets;
+};
+```
+*Open issue: The name `nestedStyleSheets` is up for discussion*
+
+## Considered alternatives
+
+1. [Declarative CSS Modules](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/ShadowDOM/explainer.md) are another mechanism for sharing styles between Declarative Shadow DOM and light DOM without the use of Javascript.
+
+## Open Issues
+
+1. Whether rules are applied automatically for `@sheet` definitions, or whether they need to be imported to apply. The CSS Working Group did not have a consensus.
+2. Fragment-only identifiers (without a URL) should allow inline `@sheet` references on the same document to be included globally (even within shadow roots). This wasn't brought up in the CSSWG discussions at all, but is important for DSD without requiring an external file (to avoid FOUC).
```

Comment:
Is this an open issue since this is what we are proposing?

Reply:
If I'm following the discussion correctly, this was in fact part of the resolution in the last CSSWG discussion. https://github.com/w3c/csswg-drafts/issues/5629#issuecomment-1498299448:

> RESOLVED: Accept @sheet with URL fragment referencing rule. Exact details to be in the Cascade spec

Reply:
Actually, I think I misunderstood. The CSSWG discussion did cover referencing URL fragment referencing, but I don't see anything about fragment-only references.